src/commercetools/testing/auth.py

```diff
@@ -69,3 +72,37 @@ def token(self, request):
             self.model.add_token(token)
             response = create_commercetools_response(request, json=token)
             return response
+
+    def introspect(self, request):
+        params = parse_qs(request.body.decode("utf-8"))
```

Comment:
Is the decode necessary? We don't do this in the token function above so I'm thinking this might not be needed?

Reply:
I wondered about this since I saw that `decode` wasn't used in `token()`. I removed it from the `parse_qs()` call, but I still need to use encode/decode on the `token` key and value [later on](https://github.com/labd/commercetools-python-sdk/pull/64/files#diff-ebea5b16b44a1be007eb55fe8ce05e20R93) because the parameters get fetched as byte objects when using `parse_qs()`.

Now that I think about it, `token()` should also be modified as well because when I test it, the `scope` parameter is not actually getting fetched in the current code, but rather it always defaults to the `manage_project:todo` fallback:

![Screenshot 2019-06-25 at 10 13 41](https://user-images.githubusercontent.com/2667373/60081527-3bdcdd00-9732-11e9-8ac0-838ca35258f1.png)

What do you think?

Reply:
For testing I was just using a curl call:
```
curl http://localhost:8989/oauth/token -X POST \
        --basic --user "{abc}:{123}" \
        -d "grant_type=client_credentials&scope=view_products:{ml-services-test-2}"
```

Reply:
I figured out the correct way to get the token without encoding shenanigans, use the following:

`token = request.qs.get('token', [None])[0]`

`request` is a `_RequestObjectProxy` object: https://github.com/jamielennox/requests-mock/blob/master/requests_mock/request.py

As for the scope, this is indeed slightly more complicated. To fix this the following should be done (but maybe out of the scope for this PR!):

- Create an API client through the mock api
- Verify in `auth.py` if the API client has a valid `client_id` and `client_secret`. 
 Right now it requires diving into internals via: 
`self.model._storage._stores['api-client']`. This contains all api-clients data created through the API.
- Return the scope from the existing API client

Reply:
>`request` is a `_RequestObjectProxy` object
Much cleaner to get using the properties. I've updated my code, thanks.

>As for the scope, this is indeed a little bit extra work. To fix this the following should be done (but maybe out of the scope for this PR!)
I suppose this does depend on the scope set for the api_client. If I have some time I can maybe have a deeper look and do a second PR for this.

Reply:
Cool, looks good :) I will merge this, check if I can refactor the auth.py and then release it!